pkg/chartverifier/checks/checks_test.go

```diff
@@ -369,22 +369,26 @@ func TestImageCertify(t *testing.T) {
 		numErrors   int
 	}
 
-	negativeTestCases := []testCase{
-		{description: "chart-0.1.0-v3.valid.tgz check images fails", uri: "chart-0.1.0-v3.valid.tgz", numErrors: 2},
+	testCases := []testCase{
+		{description: "chart-0.1.0-v3.valid.tgz check images passes", uri: "chart-0.1.0-v3.valid.tgz", numErrors: 0},
 		{description: "Helm check images fails", uri: "chart-0.1.0-v3.with-crd.tgz", numErrors: 2},
 		{description: "Helm check images fails", uri: "chart-0.1.0-v3.with-csi.tgz", numErrors: 1},
 	}
 
-	for _, tc := range negativeTestCases {
+	for _, tc := range testCases {
 		t.Run(tc.description, func(t *testing.T) {
 			config := viper.New()
 			r, err := ImagesAreCertified(&CheckOptions{URI: tc.uri, Config: config})
 			require.NoError(t, err)
 			require.NotNil(t, r)
-			require.False(t, r.Ok)
-			for i := 0; i < tc.numErrors; i++ {
-				require.Contains(t, r.Reason, ImageNotCertified)
-				r.Reason = strings.Replace(r.Reason, "_replaced_", ImageNotCertified, 1)
+			if tc.numErrors == 0 {
+				require.True(t, r.Ok)
+			} else {
+				require.False(t, r.Ok)
+				for i := 0; i < tc.numErrors; i++ {
+					require.Contains(t, r.Reason, ImageNotCertified)
+					r.Reason = strings.Replace(r.Reason, "_replaced_", ImageNotCertified, 1)
```

Comment:
I don't understand this `for` loop, since the `i` variable isn't used inside the block, and `r.Reason` is unlikely to have the string `_replaced_`; in the case it does exist, then in the case of multiple errors the reason will fix itself by replacing that string with the value of `ImageNotCertified`.

Could someone shed a light on this?

Reply:
Also, `require.Contains` stop the current test, meaning the replacement won't be even tried, which makes me think the line can be safely removed.

Reply:
Looks like the replace is wrong, it should be 
```strings.Replace(r.Reason, ImageNotCertified, "_replaced_",1)```

This is testing image certification and one error will be returned for each image which fails certification. NumErrors defines how many are expected. The replace should overwrite the one that is found so that the next loop does not find the same one again. Will fix the replace. Let me know if you want other changes, not sure how you loop numErrors times without i.

Reply:
I updated the replace invocation.